modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java

```diff
@@ -305,21 +314,18 @@ private String logErrorsToConsole(ProjectResultReport result) {
 
         List<String> messages = new ArrayList<>();
 
-        for (TestSuiteResultReport testSuiteResultReport : result.getTestSuiteResultReports()) {
-            for (TestCaseResultReport testCaseResultReport : testSuiteResultReport.getTestCaseResultReports()) {
-                for (TestStepResultReport stepResultReport : testCaseResultReport.getTestStepResultReports()) {
-                    if (stepResultReport.getAssertionStatus() == TestStepResultReport.AssertionStatusEnum.FAILED) {
-                        getLog().error("Failed " + testSuiteResultReport.getTestSuiteName() + " / " +
-                                testCaseResultReport.getTestCaseName() + " / " + stepResultReport.getTestStepName());
-                        for (String message : stepResultReport.getMessages()) {
-                            messages.add(message);
-                            getLog().error("- " + message);
-                        }
-                    }
-                }
-            }
-        }
-
+        result.getTestSuiteResultReports().stream()
```

Comment:
I agree, it was easy to understand but then sonar complains about complexity and to avoid more than 3 level of nesting. The new code is definitely not straight forward to understand, it will require fair knowledge of lambdas and streams. But this is future :)